crypto/asn1/asn_mstbl.c

```diff
@@ -72,6 +72,8 @@ static int do_tcreate(const char *value, const char *name)
         goto err;
     for (i = 0; i < sk_CONF_VALUE_num(lst); i++) {
         cnf = sk_CONF_VALUE_value(lst, i);
+        if (cnf->value == NULL)
+            continue;
         if (strcmp(cnf->name, "min") == 0) {
```

Comment:
I wonder if there is a reason why it doesnt compare min & max..

Reply:
It also appears if you have multiple keys that are the same, that it choses the last one.

Reply:
It doesn't compare min and max on the value of a key/value tuple, because X509V3_parse_list doesn't parse those values.  It could, but currently X509V3_parse_list effectively expects  a regex of the form .*:.*[,.*:.*,}...

If that pattern isn't found, the value of the key / value pair is set to NULL.  the parsing function could certainly do with a re-write to be better hardened, but given that its used in 13 call sites, I didn't want this fix to have implications throughout the code base that we weren't prepared with tests for.

And yes, mulutiple keys with the same name will get set to the last value here, which I think is sane from a user expectation standpoint, but could probably stand to be documented

Reply:
ping @slontis, any thoughts on the above?  Is it worth doing a rewrite of the X509V3_parse _list here, or is the provided isloated solution better?